backend/src/routes/requests.routes.ts

```diff
@@ -0,0 +1,89 @@
+import { Router } from "express"
+import { authenticate } from "../middleware/auth.middleware"
+import { RequestsService } from "../services/requests.service"
+import { HttpStatus } from "../constants/httpResponse"
+
+const router = Router()
+
+router.get("/leave", authenticate, async (req, res) => {
+  try {
+    const service = new RequestsService(req.app.locals.prisma)
+
+    if (!req.userId) {
+      return res.status(HttpStatus.UNAUTHORIZED).json({
+        success: false,
+        error: "User not authenticated",
+      })
+    }
+
+    const leaveRequests = await service.getLeaveRequests(req.userId)
+    res.json({
+      success: true,
+      data: {
+        ...leaveRequests,
+      },
```

Comment:
i need to learn more about the spread operator, but my trial and error shows that here we're spreading the array into one object with keys to represent the diff entries -> lets maintain the array DS here for our FE to .map function through it 🔥 

```suggestion
      data: leaveRequests,
```